api/src/websocket/handlers/subscribe.ts

```diff
@@ -0,0 +1,253 @@
+		if (!this.subscriptions[collection]) {
+			this.subscriptions[collection] = new Set();
+		}
+		this.subscriptions[collection]?.add(subscription);
+	}
+	/**
+	 * Remove a subscription
+	 * @param subscription
+	 */
+	unsubscribe(client: WebSocketClient, uid?: string) {
+		if (uid !== undefined) {
+			const subscription = this.getSubscription(uid);
+			if (subscription && subscription.client === client) {
+				this.subscriptions[subscription.collection]?.delete(subscription);
+			}
+		} else {
+			for (const key of Object.keys(this.subscriptions)) {
+				const subscriptions = Array.from(this.subscriptions[key] || []);
+				for (let i = subscriptions.length - 1; i >= 0; i--) {
+					const subscription = subscriptions[i];
+					if (!subscription) continue;
+					if (subscription.client === client && (!uid || subscription.uid === uid)) {
+						this.subscriptions[key]?.delete(subscription);
+					}
+				}
+			}
+		}
+	}
+	/**
+	 * Dispatch event to subscriptions
+	 */
+	async dispatch(event: WebSocketEvent) {
+		const subscriptions = this.subscriptions[event.collection] ?? new Set();
+		for (const subscription of subscriptions) {
+			const { client } = subscription;
+			try {
+				client.accountability = await refreshAccountability(client.accountability);
+				const schema = await getSchema();
+				const result =
+					'item' in subscription
+						? await this.getSinglePayload(subscription, client.accountability, schema, event)
+						: await this.getMultiPayload(subscription, client.accountability, schema, event);
+				client.send(fmtMessage('subscription', result, subscription.uid));
+			} catch (err) {
+				handleWebsocketException(client, err, 'subscribe');
+			}
+		}
+	}
+	/**
+	 * Handle incoming (un)subscribe requests
+	 */
+	async onMessage(client: WebSocketClient, message: WebSocketSubscribeMessage) {
+		if (getMessageType(message) === 'subscribe') {
+			try {
+				const collection = String(message.collection!);
+				const accountability = client.accountability;
+				const schema = await getSchema();
+				if (!accountability?.admin && !schema.collections[collection]) {
+					throw new WebSocketException(
+						'subscribe',
+						'INVALID_COLLECTION',
+						'The provided collection does not exists or is not accessible.',
+						message.uid
+					);
+				}
+
+				const subscription: Subscription = {
+					client,
+					collection,
+				};
+				if ('query' in message) {
+					subscription.query = sanitizeQuery(message.query!, accountability);
+				}
+				if ('item' in message) subscription.item = String(message.item);
+				if ('uid' in message) subscription.uid = String(message.uid);
+				// remove the subscription if it already exists
+				this.unsubscribe(client, subscription.uid);
+
+				let data: Record<string, any>;
+				if ('item' in subscription) {
+					data = await this.getSinglePayload(subscription, accountability, schema);
+				} else {
+					data = await this.getMultiPayload(subscription, accountability, schema);
+				}
```

Comment:
I believe this if-else is similar in nature with the ternary here (aside from the optional 4th param `event`):

https://github.com/directus/directus/blob/a9fe1ad6c818c0a5120d8a90b6f7efbbc31923b5/api/src/websocket/handlers/subscribe.ts#L139-L142

would it be possible to make both consistently if-else, or consistently ternary logic? Personally am leaning towards ternary since `data` isn't mutated here, if I'm not mistaken, but no qualms with if-else 👍